libraries/chain/block_trace.cpp

```diff
@@ -11,29 +11,53 @@
 namespace eosio { namespace chain {
 
    void shard_trace::finalize_shard() {
-      static const size_t GUESS_ACTS_PER_TX = 10;
+      {
+         static const size_t GUESS_ACTS_PER_TX = 10;
+         vector<digest_type> action_roots;
+         cpu_usage = 0;
+         action_roots.reserve(transaction_traces.size() * GUESS_ACTS_PER_TX);
+         for (const auto& tx :transaction_traces) {
+            for (const auto& at: tx.action_traces) {
+               digest_type::encoder enc;
 
-      vector<digest_type> action_roots;
-      cpu_usage = 0;
-      action_roots.reserve(transaction_traces.size() * GUESS_ACTS_PER_TX);
-      for (const auto& tx :transaction_traces) {
-         for (const auto& at: tx.action_traces) {
-            digest_type::encoder enc;
+               fc::raw::pack(enc, at.receiver);
+               fc::raw::pack(enc, at.act.account);
+               fc::raw::pack(enc, at.act.name);
+               fc::raw::pack(enc, at.act.data);
+               fc::raw::pack(enc, at.region_id);
```

Comment:
Some of this data ( like `region_id` `cycle_index`) is now duplicated between `action_trace` and `transaction_trace`.  

It seems like we can pull that data from the `transaction_trace` here and remove it from `action_trace` so that the trace structure is more concise. 

We should probably consider making sure things like this that are 64bit in the transaction merkle are 64bit in the action merkle as well.